File: pokemon-breeders-backend/src/main/java/net/tcsm/pokemonbreeders/dto/PokemonNameSearchDTO.java
package net.tcsm.pokemonbreeders.dto;

public class PokemonNameSearchDTO {

    private String name;
    private Long id;
    private String genus;
    private String identifierName;

    public PokemonNameSearchDTO(String name, Long id, String genus, String identifierName) {
        this.name = name;
        this.id = id;
        this.identifierName = identifierName;
        this.genus = genus;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getIdentifierName() {
        return identifierName;
    }

    public void setIdentifierName(String identifierName) {
        this.identifierName = identifierName;
    }

    public String getGenus() {
        return genus;
    }

    public void setGenus(String genus) {
        this.genus = genus;
    }
}

File: pokemon-breeders-frontend/src/app/pokemon-picker/pokemon-name-search-dto.ts

export class PokemonNameSearchDTO {

    constructor(name : string, id : number, genus: string, identifierName : string){
        this.name = name;
        this.id = id;
        this.genus = genus;
        this.identifierName = identifierName;
    }

    name!: string;
    id!: number;
    genus: string | undefined;
    identifierName!: string;

}